server/api/translations.js

```diff
@@ -0,0 +1,45 @@
+const express = require("express");
+const passport = require("passport");
+const translationDb = require("../services/database/translations");
+const documentDb = require("../services/database/documents");
+const { INTERPRETER } = require("../auth/roles");
+const router = express.Router();
+
+router.post(
+  "/",
+  passport.authenticate("jwt", { session: false }),
+  (req, res) => {
+    console.log(req.body);
+    const start_date = new Date();
+    const user_id = req.user.id;
+    const document_id = req.body.document_id;
+
+    const role = req.user.role;
+
+    const newTranslation = {
+      user_id,
+      document_id,
+      start_date
+    };
+  
+
+    if (role !== INTERPRETER) {
+      res.status(400).send("error");
+    } else {
+      const getTranslations = translationDb.getTranslationByDocumentId(
+        document_id
+      );
+      getTranslations.then(translations => {
+        if (translations.length !== 0) {
+          res.status(400).send("error");
+        } else {
+          translationDb.createTranslation(newTranslation);
+          documentDb.updateDocumentStatusById("Processing", document_id);
+          res.send("The translation record has been created");
```

Comment:
createTranslation and updteDocumentStatusById return promises. If you write the code like this, your API will return the success message even if the database is not updated yet. You should chain promises together to be sure you return the message to the client only after the database is updated. Try something like this:
```
translationDb.createTranslation(newTranslation)
.then(() => documentDb.updateDocumentStatusById("Processing", document_id))
.then(() => res.send("The translation record has been created"))
.catch(error => res.status(400).send("An error occurred"));
```